src/test/java/org/openrewrite/staticanalysis/RemoveUnusedParamsTest.java

```diff
@@ -0,0 +1,328 @@
+                      // no use
+                  }
+              }
+              """,
+            """
+              public class Test {
+                  static void helper() {
+                      // no use
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeUnusedConstructorParam() {
+        rewriteRun(
+          java(
+            """
+              public class Test {
+                  Test(String unused) {
+                      // ctor body
+                  }
+              }
+              """,
+            """
+              public class Test {
+                  Test() {
+                      // ctor body
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void removeUnusedVarargs() {
+        rewriteRun(
+          java(
+            """
+              public class Test {
+                  void m(String... args) {
+                      System.out.println("no args used");
+                  }
+              }
+              """,
+            """
+              public class Test {
+                  void m() {
+                      System.out.println("no args used");
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void preserveAnnotatedParamButRemoveOthers() {
+        rewriteRun(
+          java(
+            """
+              public class Test {
+                  void method(@Deprecated String keep, int removeMe) {
+                      System.out.println(keep);
+                  }
+              }
+              """,
+            """
+              public class Test {
+                  void method(@Deprecated String keep) {
+                      System.out.println(keep);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void doNotRemoveNativeMethodParam() {
+        rewriteRun(
+          java(
+            """
+              public class Test {
+                  native void nativeCall(int mustStay);
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void interfaceMethodUnchanged() {
+        rewriteRun(
+          java(
+            """
+              interface I {
+                  void foo(String param);
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void skipWronglyDueToSignatureCollision() {
+        rewriteRun(
+          java(
+            """
+            class Base {
+                void foo(int a, String b) {}
+            }
+            class Derived extends Base {
+                @Override
+                void foo(int a, String b) {}
+                void foo(String a, int b) {
+                    // no use of a or b
+                }
+            }
+            """,
+            """
+            class Base {
+                void foo(int a, String b) {}
```

Comment:
```suggestion
                public Base() {}
```
new test case - cover constructor as well.

Reply:
Addressed.

Reply:
thx